src/AsmResolver/Shims/MemoryMappedFileShim.Unix.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.ComponentModel;
+using System.Runtime.InteropServices;
+
+namespace AsmResolver.Shims;
+
+internal sealed unsafe partial class MemoryMappedFileShim
+{
+    private const int SEEK_END = 2;
+    private const int PROT_READ = 1;
+    private const int MAP_PRIVATE = 2;
+
+    [DllImport("libc", ExactSpelling = true, CharSet = CharSet.Ansi, SetLastError = true)]
+    private static extern int open([In] string path, uint flags);
+
+    [DllImport("libc", ExactSpelling = true, SetLastError = true)]
+    [return: MarshalAs(UnmanagedType.U1)]
+    private static extern bool close(int fd);
+
+    [DllImport("libc", ExactSpelling = true, SetLastError = true)]
+    private static extern nint mmap(void* addr, nuint length, int prot, int flags, int fd, long offset);
+
+    [DllImport("libc", ExactSpelling = true, SetLastError = true)]
+    [return: MarshalAs(UnmanagedType.U1)]
+    private static extern bool munmap(void* adrr, nuint length);
+
+    [DllImport("libc", ExactSpelling = true, SetLastError = true)]
+    private static extern long lseek(int fd, long offset, int whence);
+
+    private void AllocateFileUnix(string path)
+    {
+        int fd = open(path, 0);
+        if (fd == -1)
+            throw new Win32Exception();
```

Comment:
Question, do `Win32Excepion`s map onto Linux errors as well?

Reply:
Yes in CoreCLR (and newer Mono by proxy). Legacy Mono (aka Unity) does whatever https://github.com/Unity-Technologies/mono/blob/e5c9e6e3feab0841f26dd6df5e19c0fe5ab0105d/mcs/class/System/ReferenceSources/Win32Exception.cs#L9 is, which I don't know is right.